### DISCUSSION TITLE
another day, another testing issue...

### DISCUSSION THREAD

**Discussion by liebeckm** [neuefische/questions#559] [Web]
Hey @ all,

I wrote a new test for the new link in my app. I tried to keep the test close to the List-test (added an array), anyways it's not working. Even worse, since I used "useRouter" in my List, the Image test from user story 1 is also not working any longer and I have no idea whats going on. 

[Here ](https://github.com/liebeckm/capstone-project/tree/detail-page)is my Code
Here are my user stories [1](https://github.com/liebeckm/capstone-project/issues/1) and [2](https://github.com/liebeckm/capstone-project/issues/2)

Thank you! 🙏

**Comment by PReichetanz** [comment]
Hi Melissa,

could you please provide 
- the links to the corresponding files (I am not sure where the problem happens) and
- the error message of the tests?

Without these information, it's quite hard to reconstruct the error and help properly.

**Reply by liebeckm** [reply]
Hey Paul,

two tests throw error messages;
1. [List ](https://github.com/liebeckm/capstone-project/blob/detail-page/components/List/List.js) and [List Test](https://github.com/liebeckm/capstone-project/blob/detail-page/components/List/List.test.js); Error message:  NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted
2.[Link.styled.js](https://github.com/liebeckm/capstone-project/blob/detail-page/components/Link/Link.styled.js) and [Link.test.js](https://github.com/liebeckm/capstone-project/blob/detail-page/components/Link/Link.test.js) ; Error message: expect(received).toHaveBeenCalledWith(...expected)

    Matcher error: received value must be a mock or spy function

    Received has type:  object
    Received has value: <a class="sc-12c1a7bb-0 jIgKea" href="./detail-ride/undefined">Auswählen</a>

**Reply by PReichetanz** [reply]
Thank you, Melissa!

## Question 1
Your list is rendered on `pages/index.js`, so you don't need the `slug` and the `next/router` in the `List` component at all. You can simply remove the `useRouter` in the `List.js` and the test of `List.test.js` passes.


## Question 2
We don't need to test whether the next/link redirects to the correct page because this is something Next.js covers for us. This is why the only thing you can test here is that the `href` has the correct attribute (and thus link to the correct page).

You can test this in the following way:
```js
test('has the correct href attribute.', () => {
  render(
    <StyledLink href={`/detail-ride/${testRides[0].slug}`}>
      Auswählen
    </StyledLink>
  );

  const Link = screen.getByRole('link');
  expect(Link).toHaveAttribute('href', '/detail-ride/ride-1');
});
```

**Reply by liebeckm** [reply]
Thank you! Now it works :)
